clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslServer.java

```diff
@@ -159,10 +159,15 @@ private byte[] process(String tokenValue, String authorizationId, SaslExtensions
         }
         OAuthBearerToken token = callback.token();
         if (token == null) {
-            errorMessage = jsonErrorResponse(callback.errorStatus(), callback.errorScope(),
+            String returnedMessage = jsonErrorResponse(callback.errorStatus(), callback.errorScope(),
                     callback.errorOpenIDConfiguration());
+            if (!authorizationId.isEmpty()) {
+                errorMessage = String.format("The authorizationId {%s}: %s", authorizationId, returnedMessage);
+            } else {
+                errorMessage = returnedMessage;
+            }
```

Comment:
The `returnedMessage` is the standard json data for response. The `errorMessage` is the message to throw in exception, so adding authorizationId is provided.